range_queries/sparse_table.cpp

```diff
@@ -0,0 +1,54 @@
+// Implementation of Sparse Table
+//
+// Running Time Complexity
+// Build : O(NlogN)
+// Range Query : O(1)
+
+
+#include<iostream>
+#include<algorithm>
+
+const int N = 100005;
+const int LOGN = 20;
+
+int n;                  // length of the array
+int A[N];
+int logs[N];            // logs[i] = maximum p s.t. 2^p <= i
+int table[LOGN][N];     // table[i][j] = [j, j+2^i]
```

Comment:
Don't use global functions/macros.
They are not good and can lead to memory leaks.

Pass them as local function parameters instead.